### PR TITLE
Enh: Distributed agents

### PR DESCRIPTION
**Pull request by maffettone**
Moving the code from [MMM](https://github.com/NSLS-II-PDF/mmm-experiments) to Bluesky in a more robust manner. I've held off on complete adjudicator functionality for now, but will add in a future PR. 

@tacaswell Can you give this a once over to make sure it's on par with what you were expecting? Or would you prefer to hold off until some of the example docs and tests are done? 

- [x] Unit tests for kafka communications 
- [x] Address the concern of the Dispatcher not raising any errors with an open try statement.
- [x] Unit tests for race conditions and overflow of comms (moot with AgentManager plans)
- [x] Basic Mixins
  - [x] Unit test for simple mixins
  - [x] Unit tests for tiled coms
  - [x] Unit tests for plans on queue
- [x] Docs on Docs on Docs
  - [x] Extending this to NSLS-II specific example

### CODE REVIEW

**Comment by maffettone** [comment]
Note: Following this discussion, https://github.com/bluesky/bluesky-kafka/pull/50, I may remove the auto-compose from the testing and leave that up the the developer/gha to spin up as needed.

File: docs/source/conf.py [hunk 1]
```diff
@@ -86,15 +88,15 @@
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
```
Comment (maffettone) [inline docs/source/conf.py:89]: This is the only line that needed changing, but my pre-commit has an attitude problem.

**Comment by maffettone** [comment]
@dmgav It'd be good to have your eyes on this, so you know what I'm thinking from the distributed agent perspective.

**Review by maffettone — COMMENTED** [review]
This is a great change. 
I tried to follow some of the slack conversation around spinning up qserver/httpserver and the networking, and the motivation makes sense.

**Comment by tacaswell** [comment]
API comments:
 - move any work where we take a string and build something out of the `__init__`.  We may still want a class method that takes strings, but the ability to directly pass in instatiated instances is very useful (like I did for the QS object)
 - merge the methods that return the plan args and kwargs into one that returns `(plan_name, [args], {kwargs})`.  This means that agents now also have the power to select their plan and the logic for generating args and kwargs is in one place.

**Comment by maffettone** [comment]
@tacaswell This is ready for re-review. The API was updated on those two points, along with the relevant tests and docs.

**Review by dmgav — APPROVED** [review]
Looks good to me.

**Pull request merged**